semeio/jobs/ots/ots.py

```diff
@@ -0,0 +1,597 @@
+                ts = []
+                for iv in range(len(vintage_pairs.ts)):
+                    ts.append(tshift_ts[iv][point])
+                for iv in range(len(vintage_pairs.ts_simple)):
+                    ts.append(tshift_ts_simple[iv][point])
+                for iv in range(len(vintage_pairs.dpv)):
+                    ts.append(tshift_dpv[iv][point])
+                for iv, _ in enumerate(vintage_pairs.ts_rporv):
+                    ts.append(tshift_ts_rporv[iv][point])
+                f.write(line.format(xy[0], xy[1], surface_horizon[point], *ts))
+
+
+class OTS(object):
+    def __init__(
+        self,
+        eclbase,
+        mapaxes,
+        seabed,
+        youngs,
+        poisson,
+        rfactor,
+        convention,
+        above,
+        velocity_model,
+        verbose=False,
+    ):
+        """
+        The OTS class manages the information required to calculate
+        overburden timeshift.
+
+        The constructor will look for the Eclipse files INIT, EGRID
+        and UNRST based on the input case, if some of the files are
+        missing an exception will be raised. It will then instantiate
+        a EclSubsidence object will be used to manage the rest of the
+        overburden timeshift calculations.
+        """
+        case = os.path.splitext(eclbase)[0]
+        self._init_file = EclFile("%s.INIT" % case)
+        self._rst_file = EclFile("%s.UNRST" % case)
+        self._grid = EclGrid("%s.EGRID" % case, apply_mapaxes=mapaxes)
+
+        self.subsidence = EclSubsidence(self._grid, self._init_file)
+
+        self._seabed = seabed
+        self._youngs_modulus = youngs * 1e9
+        self._poisson_ratio = poisson
+        self._r_factor = rfactor
+        self._convention = convention
+        self._verbose = verbose
+
+        self._surface = res_surface = OTSResSurface(grid=self._grid, above=above)
+        if velocity_model is not None:
+            self._surface = OTSVelSurface(res_surface=res_surface, vcube=velocity_model)
+
+        self._restart_views = {}
+
+    def get_horizon(self):
+        return self._create_surface()
+
+    def _create_surface(self, z=None):
+        """
+        Generate irap surface
+
+        :param z: replace z values of surface
+        """
+        nx = self._surface.nx
+        ny = self._surface.ny
+        x = self._surface.x
+        y = self._surface.y
+        if z is None:
+            z = self._surface.z
+
+        xstart = np.min(x)
+        ystart = np.min(y)
+
+        if nx < 2 or ny < 2:
+            raise RuntimeError("Cannot create IRAP surface if nx or ny is <2")
+
+        xinc = (np.max(x) - xstart) / (nx - 1)
+        yinc = (np.max(y) - ystart) / (ny - 1)
+
+        surf = Surface(
+            nx=nx, ny=ny, xinc=xinc, yinc=yinc, xstart=xstart, ystart=ystart, angle=0
+        )
+
+        irap_x = np.empty(nx * ny)
+        irap_y = np.array(irap_x)
+
+        for i, s in enumerate(surf):
```

Comment:
Why enumerate over `surf` if you don't use it?

Reply:
Flake8's complaining when terating over indices, lists, etc. without enumeration, but this concrete case is from older ert-statoil times, so `s` can redundant.

Reply:
What? So if you do:

`for value in mylist:` that would be a flaw?! That makes no sense at all

Or do you mean the `for i in range(len(surf)):` would be an error?

Reply:
I changed it to `range(len(surf))` as we agreed that it performs faster.